java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -515,7 +516,8 @@ public ArrowFlightSqlClientHandler build() throws SQLException {
           clientBuilder.trustedCertificates(
               ClientAuthenticationUtils.getCertificateStream(keyStorePath, keyStorePassword));
         }
-        final FlightClient client = clientBuilder.build();
+        ClientCookieMiddleware.Factory factory = new ClientCookieMiddleware.Factory();
```

Comment:
Nit: Should use withMiddlewareFactories and do so before line 506 to be inconsistent with how we're adding middleware factories elsewhere.

Reply:
fixed